source/extensions/formatter/xfcc_value/xfcc_value.cc

```diff
@@ -0,0 +1,204 @@
+#include "source/extensions/formatter/xfcc_value/xfcc_value.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Formatter {
+
+namespace {
+
+const absl::flat_hash_set<std::string>& supportedKeys() {
+  // The keys are case-insensitive, so we store them in lower case.
+  CONSTRUCT_ON_FIRST_USE(absl::flat_hash_set<std::string>, {
+                                                               "by",
+                                                               "hash",
+                                                               "cert",
+                                                               "chain",
+                                                               "subject",
+                                                               "uri",
+                                                               "dns",
+                                                           });
+}
+
+// The XFCC header value is a comma (`,`) separated string. Each substring is an XFCC element,
+// which holds information added by a single proxy. A proxy can append the current client
+// certificate information as an XFCC element, to the end of the request’s XFCC header after a
+// comma.
+//
+// Each XFCC element is a semicolon (`;`) separated string. Each substring is a key-value pair,
+// grouped together by an equals (`=`) sign. The keys are case-insensitive, the values are
+// case-sensitive. If  `,`, `;` or `=` appear in a value, the value should be double-quoted.
+// Double-quotes in the value should be replaced by backslash-double-quote (`\"`).
+// Only need to find the first value with the key, so scan the header left-to-right.
```

Comment:
From Envoy's doc

> A proxy can append the current client certificate information as an XFCC element, to the end of the request’s XFCC header after a comma.

could you elaborate more why only the first value with the key matters, it seems we only need to think the oldest value? A potential feature is adding more commands for `XFCC_VALUE_FIRST` or `XFCC_VALUE_LAST`.